components/feature/tabs/src/main/java/mozilla/components/feature/tabs/tabstray/TabsFeature.kt

```diff
@@ -14,18 +14,23 @@ import mozilla.components.support.base.feature.LifecycleAwareFeature
 
 /**
  * Feature implementation for connecting a tabs tray implementation with the session module.
+ *
+ * @param defaultTabsFilter A tab filter that is used for the initial presenting of tabs that will be used by
+ * [TabsFeature.filterTabs] by default as well.
  */
 class TabsFeature(
     tabsTray: TabsTray,
     private val store: BrowserStore,
     tabsUseCases: TabsUseCases,
+    private val defaultTabsFilter: (TabSessionState) -> Boolean = { true },
     closeTabsTray: () -> Unit
 ) : LifecycleAwareFeature {
     @VisibleForTesting
     internal var presenter = TabsTrayPresenter(
         tabsTray,
         store,
-        closeTabsTray
+        closeTabsTray,
+        defaultTabsFilter
```

Comment:
What is a bit weird about this API is that I can provide a filter and it will not be used until I also call `filterTabs()`. Instead of having a default parameter in `filterTabs()` can we setup the presenter to use it?

Reply:
We're passing in the filter to the presenter because we need it for the initial state of the tab filter which we weren't doing before. So in the tabs tray on the app, we would present all the tabs, and then the app would invoke `filterTabs` to get only the non-private ones as we do [in r-b](https://github.com/mozilla-mobile/reference-browser/pull/1039/files#diff-145bf304b2e445da6fb25020ce03c650R37).

The idea behind the `filterTabs` is that it can use the same filter passed in from init or pass a new filter in.

I do agree this API is weird though. I'll try to look for something better than this.

Reply:
Looking into this more, I think we're doing the right thing here. What makes this diff look weird is that we had the same default param on the presenter and we're adding the same one on the feature, but we really should have **removed it** from the presenter when we put it on the feature - I'm making those changes now.